src/Analyser/MutatingScope.php

```diff
@@ -3820,6 +3838,31 @@ public function filterBySpecifiedTypes(SpecifiedTypes $specifiedTypes): self
 		foreach ($typeSpecifications as $typeSpecification) {
 			$expr = $typeSpecification['expr'];
 			$type = $typeSpecification['type'];
+
+			if ($expr instanceof IssetExpr) {
+				$issetExpr = $expr;
+				$expr = $issetExpr->getExpr();
+				$certainty = $issetExpr->getCertainty();
+
+				if ($certainty->no()) {
+					$scope = $scope->unsetExpression($expr);
+
+					if ($expr instanceof Variable) {
```

Comment:
Also I'm bothered by `instanceof Variable` again. I'd say we don't need this at all because `unsetExpression` should take care of everything. When NodeScopeResolver handles Unset_, it works even for variables without any ifs. See https://phpstan.org/r/889ea0c0-1b67-4894-b3f4-2f76ed4ed48e